### ISSUE TITLE
Unsupported content type: binary/octet-stream when using aws-sdk Lambda.invoke

### ISSUE THREAD

**Issue by jeremie-stratumn** [awslabs/aws-sam-cli#591]
After generating a toy example using `sam init` and spinning a local lambda endpoint using `sam local start-lambda`, I get the following error when trying to invoke the lambda from node using `aws-sdk`. Say my lambda function is named: HelloWorldFunction. I have the following `main.js` file:
```
const AWS = require("aws-sdk");

const lambda = new AWS.Lambda({
  endpoint: "http://127.0.0.1:3001/",
  region: "NA"
});

lambda.invoke(
  {
    FunctionName: "HelloWorldFunction",
    Payload: JSON.stringify({})
  },
  (err, res) => {
    console.log("back with", err, res);
  }
);
```
When I call `node main.js` I get the following error:
`UnsupportedMediaType: Unsupported content type: binary/octet-stream`

If I don't pass a payload in the invoke object, I get a response.

**Steps to reproduce the issue:**
1. `sam init`
2. `sam local start-lambda -t sam-app/template.yml`
3. `node main.js`

**Observed result:**
`UnsupportedMediaType: Unsupported content type: binary/octet-stream`

**Expected result:**
The response, with no error.

**Additional environment details (Ex: Windows, Mac, Amazon Linux etc)**

**Output of `sam --version`**:
0.5.0

**Comment by jfuss** [comment]
@jeremie-stratumn Thanks for the report. Looks like different SDKs send different headers, which I was not aware of. I was trying to get fancy [here](https://github.com/awslabs/aws-sam-cli/blob/23ffd0366207da32e6025600c14e2dbfadf687e1/samcli/local/lambda_service/local_lambda_invoke_service.py#L95) and constrain the request to be 'application/json'. It seems that as long as the request is json, it doesn't matter (I tried your example with invoke a deployed lambda function). 

I think we should just remove this header check. We are already checking to see if the request data is [json](https://github.com/awslabs/aws-sam-cli/blob/23ffd0366207da32e6025600c14e2dbfadf687e1/samcli/local/lambda_service/local_lambda_invoke_service.py#L85).

Would you be willing to contribute the fix (with updating all the respective tests)?

**Comment by jeremie-stratumn** [comment]
Thanks @jfuss for getting back to me on this. I would be glad to help! So it's just a matter of removing this check and fixing the tests right? I'll send you a PR.

Cheers

**Comment by jfuss** [comment]
@jeremie-stratumn That seems to be all. This will relate the checks we do and get us more inline with what the Lambda Service does. 

We should just update [this functional test](https://github.com/awslabs/aws-sam-cli/blob/develop/tests/functional/local/lambda_service/test_local_lambda_invoke.py#L276) to pass, which should cover the use-case you ran into with the node sdk and help catch future regressions in the future.

If you need help in some way, feel free to ping me on slack or the #samdev channel.

**Comment by jfuss** [comment]
Waiting for release, updating labels.

Thanks again @jeremie-stratumn for the PR :)

**Comment by jeremie-stratumn** [comment]
No worries!

**Comment by jfuss** [comment]
Closing this as it was released in [0.6.0](https://github.com/awslabs/aws-sam-cli/releases/tag/v0.6.0)
